--- a/buildtools/third_party/libc++/trunk/include/thread
+++ b/buildtools/third_party/libc++/trunk/include/thread
@@ -435,7 +435,7 @@
     using namespace chrono;
     if (__d > duration<_Rep, _Period>::zero())
     {
-        _LIBCPP_CONSTEXPR duration<long double> _Max = nanoseconds::max();
+        duration<long double> _Max = nanoseconds::max();
         nanoseconds __ns;
         if (__d < _Max)
         {
