diff --git a/third_party/webrtc/rtc_base/system/arch.h b/third_party/webrtc/rtc_base/system/arch.h
index be2367b85f..0e3f20d76e 100644
--- a/third_party/webrtc/rtc_base/system/arch.h
+++ b/third_party/webrtc/rtc_base/system/arch.h
@@ -50,6 +50,7 @@
 #define WEBRTC_ARCH_BIG_ENDIAN
 #endif
 #elif defined(__PPC__)
+#define WEBRTC_ARCH_PPC_FAMILY
 #if defined(__PPC64__)
 #define WEBRTC_ARCH_64_BITS
 #else
