--- a/src/DEPS
+++ b/src/DEPS
@@ -403,15 +403,15 @@ deps = {
       'packages': [
         {
           'package': 'infra/tools/luci/isolate/${{platform}}',
-          'version': 'git_revision:bc125484b8513898f17bc2501ac5e95330f44a3b',
+          'version': 'git_revision:ed999c8ac210539ae9d71f65436a2c47015712d5',
         },
         {
           'package': 'infra/tools/luci/isolated/${{platform}}',
-          'version': 'git_revision:bc125484b8513898f17bc2501ac5e95330f44a3b',
+          'version': 'git_revision:ed999c8ac210539ae9d71f65436a2c47015712d5',
         },
         {
           'package': 'infra/tools/luci/swarming/${{platform}}',
-          'version': 'git_revision:bc125484b8513898f17bc2501ac5e95330f44a3b',
+          'version': 'git_revision:ed999c8ac210539ae9d71f65436a2c47015712d5',
         },
       ],
       'dep_type': 'cipd',
