diff --git a/third_party/closure_compiler/compiler.py b/third_party/closure_compiler/compiler.py
index f09b7dd7a0ce..5308da5a64d2 100755
--- a/third_party/closure_compiler/compiler.py
+++ b/third_party/closure_compiler/compiler.py
@@ -21,7 +21,7 @@ class Compiler(object):
   and produce minified output."""

   _JAR_COMMAND = [
-      _JAVA_PATH,
+      "java ",
       "-jar",
       "-Xms1024m",
       "-client",
